src/rp2040/hardware_structs/include/hardware/structs/adc.h

```diff
@@ -1,27 +1,86 @@
+    // 0x00007000 [12-14] : AINSEL (0): Select analog mux input
+    // 0x00000400 [10]    : ERR_STICKY (0): Some past ADC conversion encountered an error
+    // 0x00000200 [9]     : ERR (0): The most recent ADC conversion encountered an error; result is...
+    // 0x00000100 [8]     : READY (0): 1 if the ADC is ready to start a new conversion
+    // 0x00000008 [3]     : START_MANY (0): Continuously perform conversions whilst this bit is 1
+    // 0x00000004 [2]     : START_ONCE (0): Start a single conversion
+    // 0x00000002 [1]     : TS_EN (0): Power on temperature sensor
+    // 0x00000001 [0]     : EN (0): Power on ADC and enable its clock
     io_rw_32 cs;
-    io_rw_32 result;
+
+    _REG_(ADC_RESULT_OFFSET)
+    // Result of most recent ADC conversion
+    // 0x00000fff [0-11]  : RESULT (0)
+    io_ro_32 result;
+
+    _REG_(ADC_FCS_OFFSET)
+    // FIFO control and status
+    // 0x0f000000 [24-27] : THRESH (0): DREQ/IRQ asserted when level >= threshold
+    // 0x000f0000 [16-19] : LEVEL (0): The number of conversion results currently waiting in the FIFO
+    // 0x00000800 [11]    : OVER (0): 1 if the FIFO has been overflowed
+    // 0x00000400 [10]    : UNDER (0): 1 if the FIFO has been underflowed
+    // 0x00000200 [9]     : FULL (0)
+    // 0x00000100 [8]     : EMPTY (0)
+    // 0x00000008 [3]     : DREQ_EN (0): If 1: assert DMA requests when FIFO contains data
+    // 0x00000004 [2]     : ERR (0): If 1: conversion error bit appears in the FIFO alongside the result
+    // 0x00000002 [1]     : SHIFT (0): If 1: FIFO results are right-shifted to be one byte in size
+    // 0x00000001 [0]     : EN (0): If 1: write result to the FIFO after each conversion
     io_rw_32 fcs;
-    io_rw_32 fifo;
+
+    _REG_(ADC_FIFO_OFFSET)
+    // Conversion result FIFO
+    // 0x00008000 [15]    : ERR (0): 1 if this particular sample experienced a conversion error
+    // 0x00000fff [0-11]  : VAL (0)
+    io_ro_32 fifo;
+
+    _REG_(ADC_DIV_OFFSET)
+    // Clock divider
+    // 0x00ffff00 [8-23]  : INT (0): Integer part of clock divisor
+    // 0x000000ff [0-7]   : FRAC (0): Fractional part of clock divisor
     io_rw_32 div;
-    io_rw_32 intr;
+
+    _REG_(ADC_INTR_OFFSET)
+    // Raw Interrupts
+    // 0x00000001 [0]     : FIFO (0): Triggered when the sample FIFO reaches a certain level
+    io_ro_32 intr;
+
+    _REG_(ADC_INTE_OFFSET)
+    // Interrupt Enable
+    // 0x00000001 [0]     : FIFO (0): Triggered when the sample FIFO reaches a certain level
     io_rw_32 inte;
+
+    _REG_(ADC_INTF_OFFSET)
+    // Interrupt Force
+    // 0x00000001 [0]     : FIFO (0): Triggered when the sample FIFO reaches a certain level
     io_rw_32 intf;
-    io_rw_32 ints;
-} adc_hw_t;
 
-check_hw_layout(adc_hw_t, ints, ADC_INTS_OFFSET);
+    _REG_(ADC_INTS_OFFSET)
+    // Interrupt status after masking & forcing
+    // 0x00000001 [0]     : FIFO (0): Triggered when the sample FIFO reaches a certain level
+    io_ro_32 ints;
+
```

Comment:
NOOOOOO!

Reply:
O.o

Reply:
Wait, what is wrong? 😅

Reply:
blank line not between fields but b4 the end of the stuct! OCD :-)

Reply:
fixed it seems :-)